akka-typed/src/main/scala/akka/typed/internal/ActorCell.scala

```diff
@@ -319,6 +315,23 @@ private[typed] class ActorCell[T](
         else unsafe.getAndAddInt(this, statusOffset, -1)
       }
     }
+
+    try {
+      unscheduleReceiveTimeout()
+      if (!isTerminated(status))
+        process()
+      scheduleReceiveTimeout()
```

Comment:
is there a reason for scheduling the receive timeout if it's terminated? (can this be moved inside the above if?)